snuba/clickhouse/native.py

```diff
@@ -269,9 +313,9 @@ def execute_robust(
             except errors.Error as e:
                 raise ClickhouseError(e.message, code=e.code) from e
 
-    def _create_conn(self) -> Client:
+    def _create_conn(self, use_fallback_host: bool = False) -> Client:
         return Client(
-            host=self.host,
+            host=(self.host if not use_fallback_host else self.get_fallback_host()),
             port=self.port,
```

Comment:
The port can be different from the one we have on the envoy proxy. So you should also keep track of the fallback node port.